sklearn/decomposition/tests/test_pca.py

```diff
@@ -49,6 +57,59 @@ def test_pca(svd_solver, n_components):
     assert_allclose(np.dot(cov, precision), np.eye(X.shape[1]), atol=1e-12)
 
 
+@pytest.mark.parametrize("density", [0.01, 0.1, 0.30])
+@pytest.mark.parametrize("n_components", [1, 2, 10, SPARSE_MAX_COMPONENTS])
+@pytest.mark.parametrize("matrix_class", CSR_CONTAINERS + CSC_CONTAINERS)
+@pytest.mark.parametrize("svd_solver", ["arpack"])
```

Comment:
Should we introduce `svd_solver="randomized"` in another PR?

Reply:
> Should we introduce svd_solver="randomized" in another PR?

This is what I'd like to do.

This test was oringinally implemented in #24415 with support for randomized and lobpcg. But there are numerical stability issues, but I think it's worth getting just arpack through right now since the performance improvement is so great.

Reply:
For the future PR: the problem might be that the default choice of `iterated_power` is too low for such a strict test to pass. Increasing it locally (in the test) to a larger value such as 30 or 100 might make it work. I haven't tried yet.